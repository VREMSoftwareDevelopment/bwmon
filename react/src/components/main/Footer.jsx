/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@mui/material';
import { footerStyles } from '../../utils/StylesUtils';

const Footer = ({ currentTime }) => {
    const classes = footerStyles();

    return (
        <div className={classes.root}>
            <Typography data-testid="app-footer1" id="app-footer1" variant="subtitle2">
                All usage information is in gigabytes
            </Typography>
            <Typography data-testid="app-footer2" id="app-footer2" variant="subtitle2">
                This page was generated on {currentTime}
            </Typography>
        </div>
    );
};

// Stryker disable next-line all
Footer.propTypes = { currentTime: PropTypes.string.isRequired };

export default Footer;
