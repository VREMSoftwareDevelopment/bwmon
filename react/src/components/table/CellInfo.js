/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

export default class CellInfo {
    constructor(id, sortable, align, label, footer, convert) {
        this.id = id;
        this.sortable = sortable;
        this.align = align;
        this.label = label;
        this.footer = footer;
        this.convert = convert;
    }
}
