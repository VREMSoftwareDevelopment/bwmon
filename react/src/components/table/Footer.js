/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { TableHead, TableRow, TableCell } from '@mui/material';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" data-testid={prefix + '-footer'} id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                const text =
                    index === 0
                        ? 'Totals'
                        : cellInfo.footer
                          ? cellInfo.convert
                              ? cellInfo.convert(values[cellInfo.id])
                              : values[cellInfo.id]
                          : '';
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

export default Footer;
