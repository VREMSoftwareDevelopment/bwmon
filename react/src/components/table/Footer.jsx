/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { TableHead, TableRow, TableCell } from '@mui/material';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" data-testid={prefix + '-footer'} id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                let text = '';
                if (index === 0) {
                    text = 'Totals';
                } else if (cellInfo.footer) {
                    text = cellInfo.convert ? cellInfo.convert(values[cellInfo.id]) : values[cellInfo.id];
                }
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

Footer.propTypes = {
    prefix: PropTypes.string.isRequired,
    cellInfos: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            align: PropTypes.string,
            footer: PropTypes.bool,
            convert: PropTypes.func,
        })
    ).isRequired,
    values: PropTypes.object.isRequired,
};

export default Footer;
