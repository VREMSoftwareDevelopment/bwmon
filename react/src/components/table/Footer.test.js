/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { create } from 'react-test-renderer';
import Footer from './Footer';
import CellInfo from './CellInfo';

describe('Footer', () => {
    const convert = (value) => value + 'xyz';

    const cellInfos = [
        new CellInfo('IP', true, 'left', 'IP', true),
        new CellInfo('MAC', false, 'left', 'MAC', false),
        new CellInfo('download', true, 'right', 'Down', true),
        new CellInfo('upload', false, 'right', 'Up', true, convert),
        new CellInfo('total', false, 'right', 'Total', false),
    ];

    const values = {
        IP: 'IP ADDRESS',
        MAC: 'MAC ADDRESS',
        download: 12345,
        upload: 67890,
        total: 2468,
    };

    it('renders correctly', () => {
        const tree = create(<Footer prefix="test" cellInfos={cellInfos} values={values} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
