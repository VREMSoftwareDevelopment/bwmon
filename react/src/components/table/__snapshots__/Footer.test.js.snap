// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<thead
  className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
    id="test-footer"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
      scope="col"
    >
      Totals
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
      scope="col"
    />
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
      scope="col"
    >
      12345
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
      scope="col"
    >
      67890xyz
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
      scope="col"
    />
  </tr>
</thead>
`;
